tesseract_environment/src/core/environment.cpp

```diff
@@ -1075,11 +1120,24 @@ bool Environment::applyAddCommand(AddLinkCommand::ConstPtr cmd)
 
     tesseract_scene_graph::Link::ConstPtr link = cmd->getLink();
     cmd = std::make_shared<AddLinkCommand>(*link, joint);
+
+    if (!scene_graph_->addLink(*cmd->getLink(), *cmd->getJoint()))
+      return false;
+  }
+  else
+  {  // A new link and joint is being added
+    if (!scene_graph_->addLink(*cmd->getLink(), *cmd->getJoint()))
+      return false;
   }
-  std::string joint_name = cmd->getJoint()->getName();
 
-  if (!scene_graph_->addLink(*cmd->getLink(), *cmd->getJoint()))
-    return false;
+  // If Link existed remove it from collision before adding the replacing links geometry
+  if (link_exists)
+  {
+    if (discrete_manager_ != nullptr)
+      discrete_manager_->removeCollisionObject(link_name);
+    if (continuous_manager_ != nullptr)
+      continuous_manager_->removeCollisionObject(link_name);
+  }
```

Comment:
This is extremely important, and for some reason I didn't think about this being required. Is there a generic way to unit test this for all relevant commands? Maybe apply the command to an environment, initialize a seperate environment with its scene graph, and then compare the two environment's collision scenes?

Reply:
I agree. I have a few unit tests checking that the contact manager is getting updated.